src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { AppDispatch, RootState } from './store';
+import { useDispatch, useSelector } from 'react-redux';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The `useSelector.withTypes<RootState>()` syntax is incorrect. To create a typed `useSelector` hook, you should use `<TSelected>(selector: (state: RootState) => TSelected) => useSelector<RootState, TSelected>(selector)`.